spec/overcommit/hook/pre_push/protected_branches_spec.rb

```diff
@@ -39,19 +39,39 @@
 
   shared_examples_for 'protected branch' do
     context 'when push is not destructive' do
-      before do
-        pushed_ref.stub(:destructive?).and_return(false)
+      context 'and destructive_only set to false' do
+        before do
+          subject.stub(:allow_non_destructive?).and_return(true)
```

Comment:
Stubbing this helper method is not the greatest, as we're not exercising the logic within. We should be varying the configuration instead.

See https://github.com/brigade/overcommit/blob/583bf0d031d5e/spec/overcommit/hook/pre_commit/forbidden_branches_spec.rb#L6-L13 for an example.